applications/ShapeOptimizationApplication/python_scripts/custom_math.py

```diff
@@ -0,0 +1,422 @@
+# ==============================================================================
+#  KratosShapeOptimizationApplication
+#
+#  License:         BSD License
+#                   license: ShapeOptimizationApplication/license.txt
+#
+#  Main authors:    Baumgaertner Daniel, https://github.com/dbaumgaertner
+#
+# ==============================================================================
+
+# Making KratosMultiphysics backward compatible with python 2.6 and 2.7
+from __future__ import print_function, absolute_import, division
+
+# Import additional libraries
+import math
+
+# ==============================================================================
+def SafeConvertVectorToMatrix(_L):
+    if IsVector(_L):
+        return [_L]
+    else:
+        return _L
+
+# ------------------------------------------------------------------------------
+def IsVector(_X):
+    if len(_X)==0 or isinstance(_X[0],(float,int)):
+        return True
+    else:
+        return False
+
+# ------------------------------------------------------------------------------
+def IsEmpty(_A):
+    _A = SafeConvertVectorToMatrix(_A)
+
+    if len(_A)==0 or len(_A[0])==0:
+        return True
+    return False
+
+# ------------------------------------------------------------------------------
+def ZeroVector(m):
+    return [0.0 for i in range(m)]
+
+# ------------------------------------------------------------------------------
+def ZeroMatrix(m,n):
+    A = [[0.0 for i in range(m)] for j in range(n)]
+    return A
+
+# ------------------------------------------------------------------------------
+def Ones(n):
+    return [1.0 for i in range(n)]
+
+# ------------------------------------------------------------------------------
+def RowSize(_A):
+    _A = SafeConvertVectorToMatrix(_A)
+
+    if len(_A)==0:
+        return 0
+    return len(_A[0])
```

Comment:
accessing a matrix with A[col][row] seems non intuitive to me

Reply:
see comment below